src/app/products/components/product-delete-confirm-dialog/product-delete-confirm-dialog.harness.ts

```diff
@@ -0,0 +1,44 @@
+import { ComponentHarness, TestElement } from "@angular/cdk/testing";
+import { MatButtonHarness } from "@angular/material/button/testing";
+
+export class ProductDeleteConfirmDialogHarness extends ComponentHarness {
+    static hostSelector = "tk-product-delete-confirm-dialog";
+
+    getTitle = this.locatorFor("[mat-dialog-title]");
+    getBody = this.locatorFor("[mat-dialog-content] p");
+    getCancelButton = this.locatorFor(
+        MatButtonHarness.with({ text: "Cancel" })
+    );
+    getDeleteButton = this.locatorFor(
+        MatButtonHarness.with({ text: "Delete" })
+    );
+
+
+    async titleText(): Promise<string> {
+        const title = await this.getTitle();
+        return title.text();
+    }
+
+    async dialogBodyText(): Promise<string> {
+        const body = await this.getBody();
+        return body.text();
+    }
+
+    async cancelButton(): Promise<MatButtonHarness> {
+        return this.getCancelButton();
+    }
+
+    async clickCancelButton(): Promise<void> {
+        const cancelButton = await this.getCancelButton();
+        await cancelButton.click();
+    }
+
+    async deleteButton(): Promise<MatButtonHarness> {
```

Comment:
weird. why do you even need this function if you can directly get call getDeleteButton property

Reply:
For most of the cases in the comments I went through the existing code and tried to write it similar way, as I assumed this is the accepted way. It applies to the methods for i.e. clicking buttons in harnesses and situations like this above. But it's true, it's quite redundant, I'll correct it. I just assumed that the rest of the application is written in the way it should be so I took some ideas form it.